crates/cxx-qt-gen/src/generator/cpp/locking.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::generator::{
+    cpp::{fragment::CppFragment, qobject::GeneratedCppQObjectBlocks},
+    naming::qobject::QObjectName,
+};
+use indoc::formatdoc;
+use syn::Result;
+
+pub fn generate(qobject_idents: &QObjectName) -> Result<(String, GeneratedCppQObjectBlocks)> {
+    let mut result = GeneratedCppQObjectBlocks::default();
+
+    let lock_guard_mutex = "::std::lock_guard<::std::recursive_mutex>";
+    let qobject_ident = qobject_idents.cpp_class.cpp.to_string();
+
+    result.includes.insert("#include <mutex>".to_owned());
+
+    result.methods.push(CppFragment::Pair {
+        header: format!("[[nodiscard]] {lock_guard_mutex} unsafeRustLock() const;"),
```

Comment:
only differing part of this from a user point of view is the new `unsafeRustLock()` method, the idea is that templated methods could use this later (but I haven't got them working yet).

But maybe there is a usecase for wanting to lock a Rust obj via C++ anyway and it's not too bad to make it a public method.

Reply:
Although it's looking like the templated connect for signals might not be possible, so maybe having this as a private method is better for now. I'll switch it to that, then we can make it public when we need it...

Reply:
Done, made it private for now.